### ISSUE TITLE
does not handle a constrained TypeVar in a type alias: assumes first constraint type

### ISSUE THREAD

**Issue by MAFulator** [microsoft/pyright#1004]
**Describe the bug**
When calling a generic method which is type annotated using a type alias
parameratized by a constrained TypeVar, the computed type in the method signature incorrectly assumes the first item from the TypeVar constraint list, rather than matching from the list.

In the example code below, in the call to `self.f_generic()` in method `f_bool()` pyright cannot match the type of the `op` arg with the signature of the generic methofd `f_generic()`. It computes the type in the signature as `Op[str]` (where `str` is the first item in the `TypeVar` constraint list) instead of  `Op[bool]`

**To Reproduce**
Copy & Paste the code below into vscode (with pylance strict mode) and observe red squiggles under the `op` arg in the call to `self.f_generic()`

**Sample Code**
```
    from typing import Callable, Generic, TypeVar

    T = TypeVar( 'T', str, bool, None )

    class MyData(Generic[T]):
        def __init__( self, val:T ):
            self.val = val

    Op = Callable[ [MyData[T]], T ]

    class MyClass:
        def f_generic( self, val:T, op:Op[T] ) -> T:
            obj = MyData[T]( val )
            return op( obj )

        def f_bool( self, val:bool ) -> bool:
            op:Op[bool] = lambda od : od.val
            r = self.f_generic( val, op )
            return r

    def main():
        o = MyClass()
        print( o.f_bool( True ) )

    if __name__ == "__main__":
        main()
```
Here is the error string from vscode on hover on the `op` error:
```
    Argument of type "Op[bool]" cannot be assigned to parameter "op" of type "Op[str]" in function "f_generic"
    Function return type "bool" is not compatible with type "str"
        Parameter 1: type "MyData[str]" cannot be assigned to type "MyData[bool]"
        "bool" is incompatible with "str"Pylance (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Running pyright implicitly via the pylance (v2020.8.0) vscode extension.

**Comment by erictraut** [comment]
Thanks for reporting the problem. This will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.67, which I just published. It will also be included in next week's release of Pylance.

**Comment by MAFulator** [comment]
Thanks Eric, you are a machine!
